generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/DefaultModelResolver.java

```diff
@@ -187,6 +189,10 @@ public ModelResolver newCopy() {
     return new DefaultModelResolver(aether, repositories, ruleNameToModelSource, modelBuilder);
   }
 
+  public VersionResolver getVersionResolver() {
+    return this.versionResolver;
+  }
```

Comment:
This is more of a pet peeve. Any way to use the default VersionResolver?

Reply:
I've made the other changes. I think this problem is reflective of a larger issue in the code base: all of these classes are very tightly coupled. In this case, the `DefaultModelResolver` has a `versionResolver` which depends on an instance of `Aether`. The `Resolver` class requires a `VersionResolver` which also needs an instance of `Aether` that needs to see the same `RemoteRepositories` as the `DefaultModelResolver`. I believe this level of refactor is outside the scope of this pull request. There are other issues here similar to this in other places.

Reply:
I am in agreement with you about the pervasive tight coupling. One precaution before investing considerable effort on that front. The bazel team is moving towards making [bazel-deps](https://github.com/johnynek/bazel-deps) the tool of choice for managing third party maven dependencies, so it may be a better use of your time to get accustomed to that tool.

Reply:
@petroseskinder OK, ty for the input.